src/library/scala/collection/IndexedSeq.scala

```diff
@@ -128,3 +131,20 @@ trait IndexedSeqOps[+A, +CC[_], +C] extends Any with SeqOps[A, CC, C] { self =>
     }
   }
 }
+
+/** A fast sliding iterator for IndexedSeqs which uses the underlying `slice` operation. */
+private final class IndexedSeqSlidingIterator[A, CC[_], C](v: IndexedSeqOps[A, CC, C], size: Int, step: Int) extends Iterator[C] {
+  private[this] val len = v.length
+  private[this] var pos = 0
+  private[this] var more = len > 0
+
+  def hasNext: Boolean = more
+
+  def next(): C = {
+    if(!more) Iterator.empty.next()
+    val slice = v.slice(pos, pos+size)
+    more = pos+step<len && pos+size<len
```

Comment:
I have no idea if it has a meaningful performance impact, but you could avoid computing `pos + size` twice (there is not technically any way to know statically that `pos` doesn't change in between, even if it never does